src/main/java/dev/j3fftw/litexpansion/resources/ThoriumResource.java

```diff
@@ -2,31 +2,41 @@
 
 import dev.j3fftw.litexpansion.Items;
 import dev.j3fftw.litexpansion.LiteXpansion;
+import io.github.thebusybiscuit.slimefun4.api.MinecraftVersion;
 import io.github.thebusybiscuit.slimefun4.api.geo.GEOResource;
+import io.github.thebusybiscuit.slimefun4.implementation.Slimefun;
+import io.github.thebusybiscuit.slimefun4.utils.biomes.BiomeMap;
+import java.util.logging.Level;
+import javax.annotation.Nonnull;
+
 import org.bukkit.NamespacedKey;
 import org.bukkit.World;
 import org.bukkit.block.Biome;
 import org.bukkit.inventory.ItemStack;
 
-import javax.annotation.Nonnull;
-import java.util.concurrent.ThreadLocalRandom;
-
 public class ThoriumResource implements GEOResource {
 
     private final NamespacedKey key = new NamespacedKey(LiteXpansion.getInstance(), "thorium");
 
+    private BiomeMap<Integer> map;
+
+    public ThoriumResource () {
+        final LiteXpansion instance = LiteXpansion.getInstance();
+
+        try {
+            if (Slimefun.getMinecraftVersion().isAtLeast(MinecraftVersion.MINECRAFT_1_18)) {
+                map = BiomeMap.getIntMapFromResource(key, instance, "/biome-maps/thorium_v1.18.json");
+            } else {
+                map = BiomeMap.getIntMapFromResource(key, instance, "/biome-maps/thorium_v1.14.json");
+            }
+        } catch (Exception e) {
+            instance.getLogger().log(Level.SEVERE, "Failed to load biome map!", e);
+        }
+    }
+
     @Override
     public int getDefaultSupply(@Nonnull World.Environment environment, Biome biome) {
-        switch (biome) {
-            case MOUNTAINS:
-            case GRAVELLY_MOUNTAINS:
-            case WOODED_MOUNTAINS:
-            case SNOWY_MOUNTAINS:
-            case MODIFIED_GRAVELLY_MOUNTAINS:
-                return ThreadLocalRandom.current().nextInt(2) + 1;
-            default:
-                return 1;
-        }
+        return map.getOrDefault(biome, 1);
```

Comment:
Every value in the biomemap is 1 anyway. So this method will return 1 or 1 either way. It will always be 1.

Reply:
updated it